src/datasets/combine.py

```diff
@@ -89,8 +119,16 @@ def interleave_datasets(
             raise ValueError(
                 f"Unable to interleave a {type(datasets[0])} with a {type(dataset)}. Expected a list of Dataset objects or a list of IterableDataset objects."
             )
+    if (iterable and stopping_strategy != "first_exhausted") or (
+        stopping_strategy not in ["first_exhausted", "all_exhausted"]
+    ):
+        raise ValueError(
```

Comment:
I think it should be a ValueError if `stopping_strategy not in ["first_exhausted", "all_exhausted"]` and a NotImplementedError if `iterable and stopping_strategy != "first_exhausted"`